app/controllers/api/v2/host_reports_controller.rb

```diff
@@ -45,9 +45,15 @@ def show
       param_group :host_report, as: :create
 
       def create
-        @host_report = HostReport.new(host_report_params)
-        if params[:host_report][:keywords].present?
-          keywords = params[:host_report][:keywords].each_with_object([]) do |n, ks|
+        the_body = params[:host_report].delete(:body)
+        if the_body && !the_body.is_a?(String)
+          logger.warn "Report body not as a string, serializing JSON"
+          the_body = JSON.pretty_generate(the_body)
+        end
```

Comment:
This part confuses me. Shouldn't you be using the `Content-Type` header to determine the content of the body?

Reply:
You are perhaps confused by the "body" term, we use it in the JSON itself for the report body, the part which is stored in "body" field in the database:

https://github.com/theforeman/smart_proxy_host_reports/blob/develop/test/snapshots/foreman-web.json#L9

The plan is to encode this whole part as JSON string-encoded field, which will save a lot of time on the Foreman side (no need to parse the JSON when importing - just take it and store it as a string).

And it worked that way, however, it is still useful to be able to send a report that does carry body as a simple JSON because it is utterly unreadable. Compare:

https://github.com/theforeman/smart_proxy_host_reports/blob/develop/test/snapshots/foreman-web.json

to

https://gist.github.com/lzap/36d7286445cdef4e47a202826c028004

So that's the idea here.